cpp/src/arrow/visit_data_inline.h

```diff
@@ -249,7 +249,7 @@ VisitArraySpanInline(const ArraySpan& arr, ValidFunc&& valid_func, NullFunc&& nu
 // The scalar value's type depends on the array data type:
 // - the type's `c_type`, if any
 // - for boolean arrays, a `bool`
-// - for binary, string and fixed-size binary arrays, a `std::string_view`
+// - for binary, string[-view] and fixed-size binary arrays, a `std::string_view`
```

Comment:
Nit: inline the other types
```suggestion
// - for binary, string, large binary and string, binary and string view, and fixed-size binary arrays, a `std::string_view`
```